### PR TITLE
fix(electron): login handling for electron

### PR DESCRIPTION
**Pull request by pengx17**

### CODE REVIEW

**Comment by pengx17** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8019?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8019** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8019?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7999** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7999?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#8032](https://github.com/toeverything/AFFiNE/pull/8032) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8032?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7994** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7994?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pengx17 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/electron/src/main/windows-manager/main-window.ts [hunk 1]
```diff
@@ -227,27 +227,50 @@ export async function showMainWindow() {
...
+        win.destroy();
+      }
+    });
+    win.on('unresponsive', reject);
+    win.on('closed', () => {
+      hiddenWindowPromise = undefined;
+    });
+    resolve(win);
+  }
+  const win = await hiddenWindowPromise;
```
Comment (forehalo) [inline packages/frontend/electron/src/main/windows-manager/main-window.ts]: I'm curious when will the hidden be closed?

Comment (pengx17) [reply packages/frontend/electron/src/main/windows-manager/main-window.ts]: it shhould be closed when the main window is closed. https://github.com/toeverything/AFFiNE/blob/f544e69d0234afa3c49fedb2f38ec64149594382/packages/frontend/electron/src/main/windows-manager/main-window.ts#L118

Comment (forehalo) [reply packages/frontend/electron/src/main/windows-manager/main-window.ts]: but it shouldn't be kept in signin progress

Comment (pengx17) [reply packages/frontend/electron/src/main/windows-manager/main-window.ts]: I think in the current workflow the hidden window will only stay in the magic link route.
A possible solution (which was implemented before but was removed) is to signal the main window when login process in the hidden window is done and then close the hidden window.

Comment (forehalo) [reply packages/frontend/electron/src/main/windows-manager/main-window.ts]: it's better to have

Comment (pengx17) [reply packages/frontend/electron/src/main/windows-manager/main-window.ts]: ok, I will add back mentioned logic

**Review by forehalo — COMMENTED** [review]
hidden window is no longer used

**Pull request closed without merging**